src/Equinox.MessageDb/MessageDb.fs

```diff
@@ -4,260 +4,112 @@ open Equinox.Core
 open Equinox.Core.Tracing
 open Equinox.MessageDb.Core
 open FsCodec
-open Serilog
 open System
-open System.Diagnostics
 open System.Text.Json
 open System.Threading
 open System.Threading.Tasks
 
 type EventBody = ReadOnlyMemory<byte>
 
-module Log =
-
-    /// <summary>Name of Property used for <c>Metric</c> in <c>LogEvent</c>s.</summary>
-    let [<Literal>] PropertyTag = "mdbEvt"
-
-    [<NoEquality; NoComparison>]
-    type Measurement = { stream: string; interval: StopwatchInterval; bytes: int; count: int }
-    [<NoEquality; NoComparison>]
-    type Metric =
-        | Slice of Measurement
-        | Batch of slices: int * Measurement
-        | ReadLast of Measurement
-        | WriteSuccess of Measurement
-        | WriteConflict of Measurement
-    let [<return: Struct>] (|MetricEvent|_|) (logEvent: Serilog.Events.LogEvent): Metric voption =
-        let mutable p = Unchecked.defaultof<_>
-        logEvent.Properties.TryGetValue(PropertyTag, &p) |> ignore
-        match p with Log.ScalarValue (:? Metric as e) -> ValueSome e | _ -> ValueNone
-
-    /// Attach a property to the captured event record to hold the metric information
-    let internal event (value: Metric) = Internal.Log.withScalarProperty PropertyTag value
-    let prop name value (log: ILogger) = log.ForContext(name, value)
-    let bytesToString (bytes: EventBody) = System.Text.Encoding.UTF8.GetString bytes.Span
-    let propEvents name (kvps: System.Collections.Generic.KeyValuePair<string, string> seq) (log: ILogger) =
-        let items = seq { for kv in kvps do yield sprintf "{\"%s\": %s}" kv.Key kv.Value }
-        log.ForContext(name, sprintf "[%s]" (String.concat ",\n\r" items))
-    let propEventData name (events: IEventData<EventBody>[]) (log: ILogger) =
-        log |> propEvents name (seq {
-            for x in events do
-                yield System.Collections.Generic.KeyValuePair<_, _>(x.EventType, bytesToString x.Data) })
-    let propResolvedEvents name (events: ITimelineEvent<EventBody>[]) (log: ILogger) =
-        log |> propEvents name (seq {
-            for x in events do
-                yield System.Collections.Generic.KeyValuePair<_, _>(x.EventType, bytesToString x.Data) })
-    let withLoggedRetries<'t> retryPolicy (contextLabel: string) (f: ILogger -> CancellationToken -> Task<'t>) log ct: Task<'t> =
+module Activity =
+    let setTags (tags: (string * obj)[]) =
+        let act = System.Diagnostics.Activity.Current
+        if act <> null then for k,v in tags do act.SetTag(k, v) |> ignore
+
+module Retry =
+    let withSpanTag<'t> retryPolicy (f: CancellationToken -> Task<'t>) ct: Task<'t> =
         match retryPolicy with
-        | None -> f log ct
+        | None -> f ct
         | Some retryPolicy ->
-            let withLoggingContextWrapping count =
-                let log = if count = 1 then log else log |> prop contextLabel count
```

Comment:
count was an attempt number, has it  changed? if so, maybe count should become `retryCount`?